main.c

```diff
@@ -3,10 +3,10 @@
 #include <string.h>
 
 #include "stackconfig.h"
-#include "stack.h"
+#include "stack.c"
```

Comment:
So doing this allows you to simply compile main.c instead of linking with stack.c. Your code looks great, but this include looks bad. Great work! Keep it up.